### DISCUSSION TITLE
Need Help : Processing a metric timestamp value to unix time epoch nanoseconds

### DISCUSSION THREAD

**Discussion by rk3094** [vectordotdev/vector#18026] [Q&A] [answered]
Hi,

I am new to vector and VRL. I have a metric that I am trying to sent to GCP cloud monitoring sink. There is a requirement from the consumer service to accept metric timestamp in unix epoch time format of nanoseconds. Here is a json of the metric that I have:
`
{"name":"some_service.metric_name_sample","tags":{"key1":"val1","key2":"val2"},"timestamp":"2023-07-19T16:56:42.792764503Z","kind":"absolute","gauge":{"value":1.0}}
`
I need to convert the timestamp field to unix epoch time format in nanoseconds. I tried to do this using a transform of "remap" type in vector remap language using `to_unix_timestamp` method. But it is not having any effect on the json. Here is the code in config.toml:

> 
> [transforms.change_timestamp_format]
>     type = "remap"
>     inputs = ["new_metrics"]
>     source = '''
>     .tags.new_time = to_unix_timestamp!(.timestamp, unit: "nanoseconds")
>     '''

Expectation is to get a new key value pair in the json blob inside tags as "new_time": "1689765925"
Can somebody please help me with this?

**Comment by jszwedko** [comment]
Can you share the full config? That looks roughly like what I would have suggested.

**Reply by rk3094** [reply]
> [sources.vector_metrics_source]
    type = "prometheus_scrape"
    endpoints = [ "http://some_endpoint:port/metrics" ]
    
    [transforms.filter_metrics]
    type = "filter"
    inputs = [ "vector_metrics_source" ]
    condition = '.name == "some_service.metric_name_sample"'
    
    [transforms.enrich_metrics]
    type = "remap"
    inputs = ["filter_metrics"]
    source = '''      
    .tags.key1= "val1"
    .tags.key2 = "val2"
    '''

    [transforms.change_timestamp_format]
    type = "remap"
    inputs = ["new_metrics"]
    source = '''
    .tags.new_time = to_unix_timestamp!(.timestamp, unit: "nanoseconds")
    '''
    
    [sinks.print]
    type = "console"
    inputs = ["change_time_format"]
    encoding.codec = "json"

**Reply by rk3094** [reply]
Vector version info:
  INFO vector: Vector has started. debug="false" version="0.31.0" arch="x86_64" revision="0f13b22 2023-07-06 13:52:34.591204470"

**Reply by jszwedko** [reply]
What is the output you are seeing?

**Reply by rk3094** [reply]
I am seeing the same output:
{"name":"some_service.metric_name_sample","tags":{"key1":"val1","key2":"val2"},"timestamp":"2023-07-19T16:56:42.792764503Z","kind":"absolute","gauge":{"value":1.0}}

There is no new_time key in the output

**Reply by jszwedko** [reply, marked as answer]
Try adding `to_string()` around the `to_unix_timestamp!(.timestamp, unit: "nanoseconds")`. I'm thinking maybe tag values need to be strings.

**Reply by rk3094** [reply]
Thank you so much @jszwedko . That solved the problem :)
